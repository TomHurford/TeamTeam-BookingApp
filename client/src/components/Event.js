import React, {Component} from "react";
import '../styles/EventCard.css';

//Creating an event component to show event details

class Event extends Component { 
  render() {
    return (
      <div className="eventCard">
        <h3 className="eventName">{this.props.specificEvent.name}</h3>
        <p className="eventDescription">{this.props.specificEvent.description}</p>
        <b><span className="eventDate">{this.props.specificEvent.date}</span></b>
        <b><span className="eventLocation">{this.props.specificEvent.location}</span></b>
      </div>
    );
  }
}

export default Event;