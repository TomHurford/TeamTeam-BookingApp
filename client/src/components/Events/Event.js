import React, {Component} from "react";
import '../../styles/EventCard.css';

//Creating an event component to show event details

class Event extends Component { 
  render() {
    return (
      <div className="eventCard" data-testid = "eventCardID">
        <h3 className="eventName" data-testid = "eventNameID">{this.props.specificEvent.name}</h3>
        <p className="eventDescription" data-testid = "eventDescriptionID">{this.props.specificEvent.description}</p>
        <b><span className="eventDate" data-testid = "eventDateID">{this.props.specificEvent.date}</span></b>
        <b><span className="eventLocation" data-testid = "eventLocationID">{this.props.specificEvent.location}</span></b>
      </div>
    );
  }
}

export default Event;