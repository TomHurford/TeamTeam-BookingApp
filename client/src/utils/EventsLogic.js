// This file contains functions that make API calls to the backend
const axios = require('axios').default;

// This function gets all events from the backend
export const getEvents = async () => {
  try{
    const response = await axios.get('http://localhost:5001/events')
    const eventsArray = await response.data;
    const events = eventsArray.events;
    return events;
  } catch (error) {
    return null;
  }
}

// This function gets a specific event from the backend
export const getEventById = async (eventId) => {
  try{
    const response = await axios.post('http://localhost:5001/events', {eventId: eventId});
    const event = await response.data;
    return event;
  } catch (error) {
    return null;
  }
}