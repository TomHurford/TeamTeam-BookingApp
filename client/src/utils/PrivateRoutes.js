import React from "react";
import { Outlet, Navigate } from "react-router-dom";
const jwtController = require("./jwt.js");

// A function for the private routes which prevents non logged in users from accessing pages they are not authorised to access.
export const PrivateRoutes = () => {
  let auth = jwtController.getToken();
  return auth ? <Outlet /> : <Navigate to="/login" />;
};

// A function for the logged in routes which prevents logged in users from accessing pages they are not authorised to access.
export const LoggedInRoutes = () => {
  let auth = !jwtController.getToken();
  return auth ? <Outlet /> : <Navigate to="/home" />;
};
