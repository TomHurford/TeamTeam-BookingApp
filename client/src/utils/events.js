// This file contains functions that make API calls to the backend

// This function gets all events from the backend
export const getEvents = async () => {
  const response = await fetch('http://localhost:5001/events', 
    {
      method: 'GET',
      mode: 'cors', 
      credentials: 'same-origin'
    }
  );
  const data = await response.json();
  const events = data.events;
  return events;
}

// This function gets a specific event from the backend
export const getEventById = async (eventId) => {
  const response = await fetch('http://localhost:5001/events', 
    {
      method: 'POST',
      mode: 'cors', 
      credentials: 'same-origin',
      headers: {
        'Content-Type': 'application/json',
      },
      body : JSON.stringify({"eventId": eventId})
    }
  );
  const event = await response.json();
  return event;
}

