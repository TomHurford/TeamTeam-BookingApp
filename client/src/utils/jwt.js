// import jwt from 'jsonwebtoken';

const axios = require('axios');
const sessionStorage = require('sessionstorage');

// export const decodeToken = (token) => {
//     return jwt.decode(token);
// }

export const getToken = () => {
    return sessionStorage.getItem('token');
}

export const setToken = (token) => {
    console.log('SET TOKEN FUNC')
    sessionStorage.setItem('token', token);
}

export const removeToken = () => {
    sessionStorage.removeItem('token');
}


// export const isTokenExpired = (token) => {
//     const decoded = decodeToken(token);
//     if (!decoded) return true;
//     return decoded.exp < Date.now() / 1000;
// }

export const checkIsLoggedIn = async () => {
    const token = getToken();

    if (token == null || token == 'null' || token == undefined) {
        return false;
    }

    const headers = {
        'Content-Type': 'application/json',
        'Authorization': 'Bearer ' + token
      }      

    const res = await axios.post('https://localhost:5001/user/checkUserLoggedIn', {}, {
        headers: headers
      }).catch(err => {
        console.log(err);
      })
    if (res !== undefined && res.status == 200) {
        return true;
    } else {
        removeToken()
    }
    return false;

}

// export const requireAuth = (nextState, replace) => {
//     if (!isLoggedIn()) {
//         replace({
//             pathname: '/login',
//             state: { nextPathname: nextState.location.pathname }
//         });
//     }
// }

// Send a request to the server to refresh the token, 
export const refreshAuthToken = () => {
    axios.post('https://localhost:5001/user/checkUserLoggedIn', { token: getToken() })
        .then(response => {
            setToken(response.data.token);
        })
        .catch(error => {
            // Redirect to login page if the token is invalid
            removeToken();
            window.location.href = '/login';
            console.log(error.message);
        });
}