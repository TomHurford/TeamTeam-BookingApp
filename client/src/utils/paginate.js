import _ from "lodash";

// This is a utility function that returns a subset of the items array
export function paginate(items, pageNumber, pageSize) {
  const startIndex = (pageNumber - 1) * pageSize;
  return _(items).slice(startIndex).take(pageSize).value();
}
