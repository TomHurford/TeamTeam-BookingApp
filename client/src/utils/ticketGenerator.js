const axios = require('axios');
const jwt = require("./jwt.js");

const generateTickets = async (event, ticketTypes, tickets, total) => {

    // Check Logged In
    var isLoggedIn = false;

    await jwt.checkIsLoggedIn().then((res) => {if (res) {isLoggedIn = res;}});

    if (!isLoggedIn) {
        return 0;
    }

    // Create tickets for each.

    if (event.event.id != ticketTypes[0].eventId) {
        return 0;
    }

    var token = jwt.getToken();

    const headers = {
        'Content-Type': 'application/json',
        'Authorization': 'Bearer ' + token
      }      

    var types = [];
    ticketTypes.map((ticketType) => {
        
        types.push({
            "id": ticketType.id,
            "quantity": tickets[ticketType.id]
        });

    });

    const res = await axios.post('http://localhost:5001/purchase/create', {
            status: "paid",
            method: "air",
            total: total,
            ticket_quantities: {
                types: types
            },
            eventId: event.event.id
        },{
            headers: headers
        }).catch(err => {
            console.log(err);
        })

    console.log(res);

    if (res.status === 200) return 1;

    return 0;
}

module.exports = {
    generateTickets
};
