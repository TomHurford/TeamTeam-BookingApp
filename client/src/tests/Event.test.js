import React from 'react';
import renderer from 'react-test-renderer';
import Event from '../components/Events/Event';
import { getEventById } from '../utils/EventsLogic';
import { getByTestId, render } from '@testing-library/react';

test('Event snapshot test', () => {
  const component = renderer.create(<Event specificEvent={getEventById(1)}/>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Event is defined', () => {
  const {getByTestId} = render(<Event specificEvent={{
  "id": 1,
  "name": "Event 1",
  "description": "Event 1 description",
  "date": "2023-12-02T00:00:00.000Z",
  "location": "Event 1 location"
  }}/>);
  const event = getByTestId('eventCardID');
  expect(event).toBeDefined();
});

test('Event component renders 4 elements', () => {
  const {getByTestId} = render(<Event specificEvent= {{
    "id": 1,
    "name": "Event 1",
    "description": "Event 1 description",
    "date": "2023-12-02T00:00:00.000Z",
    "location": "Event 1 location"
  }}/>);
  const event = getByTestId('eventCardID');
  expect(event.children.length).toBe(4);
});

test('Event name is correct', () => {
  const {getByTestId} = render(<Event specificEvent={{
    "id": 1,
    "name": "Event 1",
    "description": "Event 1 description",
    "date": "2023-12-02T00:00:00.000Z",
    "location": "Event 1 location"
  }}/>);

  const eventName = getByTestId('eventNameID');
  expect(eventName.textContent).toBe("Event 1");
});

test('Event description is correct', () => {
  const {getByTestId} = render(<Event specificEvent={{
    "id": 1,
    "name": "Event 1",
    "description": "Event 1 description",
    "date": "2023-12-02T00:00:00.000Z",
    "location": "Event 1 location"
  }}/>);
  const eventDescription = getByTestId('eventDescriptionID');
  expect (eventDescription.textContent).toBe("Event 1 description");
});

test('Event date is correct', () => {
  const {getByTestId} = render(<Event specificEvent={{
    "id": 1,
    "name": "Event 1",
    "description": "Event 1 description",
    "date": "2023-12-02T00:00:00.000Z",
    "location": "Event 1 location"
  }}/>);
  const eventDate = getByTestId('eventDateID');
  expect (eventDate.textContent).toBe("2023-12-02T00:00:00.000Z");
});

test('Event location is correct', () => {
  const {getByTestId} = render(<Event specificEvent={{
    "id": 1,
    "name": "Event 1",
    "description": "Event 1 description",
    "date": "2023-12-02T00:00:00.000Z",
    "location": "Event 1 location"
  }}/>);
  const eventLocation = getByTestId('eventLocationID');
  expect (eventLocation.textContent).toBe("Event 1 location");
});










