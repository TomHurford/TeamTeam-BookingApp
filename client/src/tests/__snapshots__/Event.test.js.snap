// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event snapshot test 1`] = `
<div
  className="eventCard"
  data-testid="eventCardID"
>
  <h3
    className="eventName"
    data-testid="eventNameID"
  />
  <p
    className="eventDescription"
    data-testid="eventDescriptionID"
  />
  <b>
    <span
      className="eventDate"
      data-testid="eventDateID"
    />
  </b>
  <b>
    <span
      className="eventLocation"
      data-testid="eventLocationID"
    />
  </b>
</div>
`;
